### ISSUE TITLE
Degraded performance

### ISSUE THREAD

**Issue by vkanithi** [sdepold/jquery-rss#127]
Hi,

We are seeing performance issues with feed going through feedrapp. This all started Monday (Mar 27).
Fiddler shows 14 sec response time, any suggestions to improve this?

Appreciate your help.
Thanks

**Comment by jeremyjburnett** [comment]
Hello, I've also noticed slow load times with feedrapp. Starting March 27 the load time has slowed to 15 sec to 20 sec.

A fix would be greatly appreciated!

Thanks

**Comment by ArmitaFarahavar** [comment]
Hi there,
We have been having issues with the load time since Monday as well. Do you have any suggestions on how to fix the load time? is there any fix?

Any help would be appreciated.
Thank you

**Comment by sdepold** [comment]
Hm interesting. Can you pls tell me the feed URL?

**Comment by vkanithi** [comment]
Here the feed that I have been using: https://www.smartbrief.com/servlet/rss?b=ACC
This had been working perfect till last week.

**Comment by sdepold** [comment]
Some context: I migrated from heroku to aws and expected some actual perf improvement but maybe I've been wrong with that assumption...

**Comment by vkanithi** [comment]
Are you planning on bringing back to Heroku?

Thanks

**Comment by sdepold** [comment]
I will think about the options over the weekend and will also debug the issue a bit.

**Comment by jeremyjburnett** [comment]
The feeds are not working at all now, have you made an update?

**Comment by sdepold** [comment]
Should be fine now. The HDD ran full. :(

**Comment by sdepold** [comment]
Alright. Can you please tell me if the issue is resolved?

**Comment by warhol15** [comment]
It is working again, for me.

**Comment by jufracaqui** [comment]
All working like a charm now. Thank you very much!

**Comment by warhol15** [comment]
Forgot to say this: thank you.

**Comment by vkanithi** [comment]
Thank You, it is working fine now.

Appreciate your help.

**Comment by ArmitaFarahavar** [comment]
It's working again and it's way faster than before. Thank you so much :)

**Comment by jeremyjburnett** [comment]
All fixed, thanks!

**Comment by sdepold** [comment]
:)
jeremyjburnett <notifications@github.com> schrieb am Mo. 3. Apr. 2017 um
19:43:

**Comment by sdepold** [comment]
:)

**Comment by vkanithi** [comment]
Looks like the issue is back again, it is slow. Can you check?

**Comment by sdepold** [comment]
I think the server is simply not capable of handling the load ... It eats up a quite significant amount of memory 🤔  Will look into it!

**Comment by sdepold** [comment]
😱  I just noticed that the generated traffic is exceeding the free traffic on AWS massively. I basically just ran the for a couple of days and it already passed the available amount of network IO for the entire month. 😭  I switched back to heroku for now. Will take further looks in the next days.

**Comment by sdepold** [comment]
FWIW: The currently served traffic is 126GB for April... I'm open for ideas.

**Comment by sdepold** [comment]
Just a tiny recap on this:

- I will look into the issue more although I'm unsure about what can be done as a next step.
- I just stopped the AWS instance to save some $$$
- I re-enabled the Heroku instance and rewired the DNS settings respectively.
- If anyone of you is interesting in the actual feedrapp project and would potentially even be able to offer some sort of free hosting solution, I'd love to hear about it!

**Comment by sdepold** [comment]
Aaaaalright. I just moved the service to a new server which seems to have no bigger issues with the load and is relatively cheap. Can you please tell me how everythings works performance-wise for you?

**Comment by jufracaqui** [comment]
Right now, at least for me, it is pretty much instant and without issues.

Thank you and we all appreciate your work and effort.

**Comment by zmarty** [comment]
It's happening again. US west coast.

**Comment by sdepold** [comment]
Should be better :)

**Comment by sdepold** [comment]
Hey everybody.

I made a rather subtle change in the feedr server which I hope has a huge impact on the stability of the server. You can find the change here: https://github.com/sdepold/feedrapp/pull/71

TLDR: I moved the caching layer far up in the request hierarchy and improved the performance of the cache hits by 100-1000 times. A cache hit was previously taking around 13ms to get served. It is now down to 0.01ms. I'd love to get some feedback about the performance from you!

**Comment by alepech** [comment]
Hello sdepold, i wants to use my own host server for run feedr. Do you have some guide or tutorial ?

"...If you don't want to rely on the provided server and instead run your own version, you can just download feedr, install the dependencies and run it..."

Where i can download feedr¿ and how i can isntall on the server?

Thank you.

**Comment by sdepold** [comment]
Pls check http://www.feedrapp.info/#hosting-section

**Comment by alepech** [comment]
Hello sdepold, there isnt a link to download the app? i dont know how to start ... i never do this =)

**Comment by sdepold** [comment]
I'll close this issue for now since everything seems to work fine. feel free to reopen
